src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.Book;
+
+public class BookDaoImpl implements BookDaoInt {
+    public static final String URL = "jdbc:mysql://localhost:3306/book";
+    public static final String USER = "root";
+    public static final String PASSWORD = "1111";
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD)) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            int updatedRows = preparedStatement.executeUpdate();
+            if (updatedRows == 0) {
+                throw new RuntimeException("Creating book failed, no rows affected.");
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Runtime exception while creating book."
+                    + " Can't create book", e);
+        } catch (SQLException e) {
+            throw new RuntimeException("SQL error while creating book."
+                    + " Can't create book", e);
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String query = "SELECT * FROM books";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD)) {
+            Statement statement = connection.createStatement();
```

Comment:
Use PreparedStatement over Statement, even for a static query with no parameters in findAll() method. It's the best practice, and it's slightly faster.

see checklist file

Reply:
Not fixed